pkg/i2gw/notifications/notifications.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package notifications
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/jedib0t/go-pretty/v6/table"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func init() {
+	CommonNotification = NotificationAggregator{map[string][]Notification{}}
+}
+
+const (
+	InfoNotification    MessageType = "INFO"
+	WarningNotification MessageType = "WARNING"
+	ErrorNotification   MessageType = "ERROR"
+)
+
+type MessageType string
+
+type Notification struct {
+	Type           MessageType
+	Message        string
+	CallingObjects []client.Object
+}
+
+type NotificationAggregator struct {
```

Comment:
lets add a lock here, and then in dispatch we should acquire the lock before we add something. Also when we read the notifications we should acquire the locks first.

We have nothing concurrent yet but this sets up a good infrastructure for the future